applications/RANSApplication/custom_processes/rans_check_scalar_bounds_process.h

```diff
@@ -0,0 +1,122 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+#if !defined(KRATOS_RANS_CHECK_SCALAR_BOUNDS_PROCESS_H_INCLUDED)
+#define KRATOS_RANS_CHECK_SCALAR_BOUNDS_PROCESS_H_INCLUDED
+
+// System includes
+#include <string>
+
+// External includes
+
+// Project includes
+#include "containers/model.h"
+#include "processes/process.h"
+
+namespace Kratos
+{
+///@addtogroup RANSApplication
+///@{
+
+///@name Kratos Classes
+///@{
+
+/**
+ * @brief Checks lower and upper bounds of a scalar
+ *
+ * This process checks lower and upper bounds of a variable in nodes in a given modelpart
+ *
+ */
+
+class KRATOS_API(RANS_APPLICATION) RansCheckScalarBoundsProcess : public Process
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    using NodeType = ModelPart::NodeType;
+
+    /// Pointer definition of RansCheckScalarBoundsProcess
+    KRATOS_CLASS_POINTER_DEFINITION(RansCheckScalarBoundsProcess);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor
+
+    RansCheckScalarBoundsProcess(
+        Model& rModel,
+        Parameters rParameters);
+
+    /// Destructor.
+    ~RansCheckScalarBoundsProcess() override = default;
+
+    /// Assignment operator.
+    RansCheckScalarBoundsProcess& operator=(RansCheckScalarBoundsProcess const& rOther) = delete;
+
+    /// Copy constructor.
+    RansCheckScalarBoundsProcess(RansCheckScalarBoundsProcess const& rOther) = delete;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    int Check() override;
+
+    void Execute() override;
+
+    void ExecuteFinalizeSolutionStep() override;
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override;
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override;
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override;
+
+    ///@}
+
+private:
+    ///@name Member Variables
+    ///@{
+
+    Model& mrModel;
+    Parameters mrParameters;
+    std::string mModelPartName;
+    std::string mVariableName;
```

Comment:
Maybe you can save references to the variable and the model part to avoid accessing them each time the process is used.

Reply:
removed process :)